### PR TITLE
Correctly document returns from methods returning JError calls

### PR DESCRIPTION
**Pull request by mbabker**
#### Summary of Changes

Ensure correct return types are documented for methods where a `return JError::raise*()` line is in use.
#### Testing Instructions

Review.

### CODE REVIEW

File: libraries/legacy/application/application.php [hunk 1]
```diff
@@ -609,7 +609,7 @@ public function getUserStateFromRequest($key, $request, $default = null, $type =
 	 * @param   array  $credentials  Array('username' => string, 'password' => string)
 	 * @param   array  $options      Array('remember' => boolean)
 	 *
-	 * @return  boolean  True on success.
+	 * @return  boolean|JException  True on success, false if failed or silent handling is configured, or a JException object on authentication error.
 	 *
```
Comment (ggppdk) [inline libraries/legacy/application/application.php]: In other file you removed the "break;" after return, maybe you would want to do it here too

**Comment by ggppdk** [comment]
I have tested this item :white_check_mark: successfully on 1eabac5c2614beb21a76a4d68ac7d7d9a27f0a84<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/11238">issues.joomla.org/joomla-cms/11238</a>.</sub>

**Comment by RonakParmar** [comment]
I have tested this item :white_check_mark: successfully on 1eabac5c2614beb21a76a4d68ac7d7d9a27f0a84<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/11238">issues.joomla.org/joomla-cms/11238</a>.</sub>

**Comment by brianteeman** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/11238">issues.joomla.org/joomla-cms/11238</a>.</sub>

**Pull request merged**